code/modules/mob/interactive.dm

```diff
@@ -0,0 +1,660 @@
+				TARGET = pick(target_filter(orange(MIN_RANGE_FIND,src)))
+			else
+				TARGET = pick(get_area_turfs(job2area(myjob)))
+		tryWalk(TARGET)
+	LAST_TARGET = TARGET
+
+/mob/living/carbon/human/interactive/proc/tryWalk(var/turf/TARGET)
+	if(!isnotfunc())
+		if(!walk2derpless(TARGET))
+			timeout++
+	else
+		timeout++
+
+
+/mob/living/carbon/human/interactive/proc/walk2derpless(var/target)
+	set background = 1
+	var/turf/T = get_turf(target)
+	var/turf/D = get_step(src,dir)
+	if(D)
+		if(!D.density)
+			walk_to(src,T)
+			doing = doing & ~TRAVEL
+			return 1
+		else
+			sidestep(D)
+			doing = doing & ~TRAVEL
+			return 0
+	else
+		doing = doing & ~TRAVEL
+		return 0
+
+/mob/living/carbon/human/interactive/proc/job2area(var/target)
+	var/datum/job/T = target
+	if(T.title == "Assistant")
+		return /area/hallway/primary
+	if(T.title == "Captain" || T.title == "Head of Personnel")
+		return /area/bridge
+	if(T.title == "Bartender" || T.title == "Chef")
+		return /area/crew_quarters
+	if(T.title == "Station Engineer" || T.title == "Chief Engineer" || T.title == "Atmospheric Technician")
+		return /area/engine
+	if(T.title == "Chief Medical Officer" || T.title == "Medical Doctor" || T.title == "Chemist" || T.title == "Virologist" || T.title == "Geneticist")
+		return /area/medical
+	if(T.title == "Research Director" || T.title == "Scientist" || T.title == "Roboticist")
+		return /area/toxins
+	if(T.title == "Head of Security" || T.title == "Warden" || T.title == "Security Officer" || T.title == "Detective")
+		return /area/security
+	else
+		return pick(/area/hallway,/area/crew_quarters)
+
+/mob/living/carbon/human/interactive/proc/target_filter(var/target)
+	var/list/L = target
+	for(var/atom/A in target)
+		if(istype(A,/area) || istype(A,/turf/unsimulated) || istype(A,/turf/space))
+			L -= A
+	return L
+
+/mob/living/carbon/human/interactive/proc/denied_filter(var/target)
+	var/list/denied = list(/obj/structure/window,/obj/structure/table) //expand me
+	for(var/a in denied)
+		if(istype(target,a))
+			return 1
+	return 0
+
+///BUILT IN MODULES
+/mob/living/carbon/human/interactive/proc/chatter(obj)
+	var/list/verbs_use = list("touch","hold","poke","punch","smash","assault","wrench","tickle","fist","modify")
+	var/list/verbs_touch = list("stroke","pat","lick","finger","mess with","absorb","sneeze on","feel up","inject")
+	var/list/verbs_move = list("run","jog","leg it","slide","cheese it","walk","stroll","fly","scoot","yakkety")
+	var/list/nouns_insult = list("faggot","shitcurity","noob","comdom","greyturd","tator","lingbin","bitch","whorelord")
+	var/list/nouns_generic = list("you", "you there", "guy", "person", "dude", "shitbin", "officer")
+	var/list/nouns_objects = list("thing","tool","whatever","potential murder weapon")
+	var/list/nouns_body = list("ass","head","face","arms","fingers","toes","chest","eyes","nails","hoo-hah")
+	var/list/adjective_insult = list("ugly","repulsive","dumb","asian","childish","captain-like","monkey-like","whoreish")

```

Comment:
It's not, it's an adjective (in this sense). I threw it in there because it creates some mildly amusing generated sentances: 
![lolwoops](https://cloud.githubusercontent.com/assets/3829126/5601582/2eb555f4-935d-11e4-8312-cb88725b2859.PNG)
If it's too offensive/racist, I can remove it.